src/dappsExplorer/TabDiscover.tsx

```diff
@@ -76,6 +82,14 @@ function TabDiscover({ navigation }: Props) {
   const nonFavoriteDappsWithCategoryNames = useSelector(nonFavoriteDappsWithCategoryNamesSelector)
   const favoriteDappsWithCategoryNames = useSelector(favoriteDappsWithCategoryNamesSelector)
 
+  // Earning Pool Aave
+  const token = useTokenInfo(networkConfig.arbUsdcTokenId)
+  const walletAddress = useSelector(walletAddressSelector)
+  const asyncPoolUserInfo = useAavePoolUserBalance({
```

Comment:
should this be conditional on `token?.address` being truthy?  The null dereference implies `assetAddress` might end up being `null` (which is not of type `Address`).

Reply:
+1, and we should avoid `as type` as much as possible in production code

Reply:
`token.address` returned from `useTokenInfo` is of type string, but the `readContract` method in `fetchAavePoolInfo` expects an `Address` type. We do a lot of `...as Address` with jumpstart so I am not sure if this is avoidable when using smart contract methods.

Reply:
This has since moved to the `EarnActivePool.tsx` component itself so we could do a check and simply return null for that component with `if(!token?.address) return null`.

Reply:
Continued here https://github.com/valora-inc/wallet/pull/5400/files#r1594881960